### PR TITLE
net-vpn/I2pd-2.27.0 version bump

### PR DESCRIPTION
**Pull request by khumarahn**
regular bump, so changes

### CODE REVIEW

File: net-vpn/i2pd/i2pd-2.27.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	newinitd "${FILESDIR}/i2pd-2.6.0-r3.initd" i2pd
+	systemd_newunit "${FILESDIR}/i2pd-2.6.0-r3.service" i2pd.service
+
+	# logrotate
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}/i2pd-2.6.0-r3.logrotate" i2pd
+}
+
+pkg_preinst() {
+	enewgroup "${I2PD_GROUP}"
```
Comment (juippis) [inline net-vpn/i2pd/i2pd-2.27.0.ebuild]: This, and user creation, is now handled via methods by https://www.gentoo.org/glep/glep-0081.html
Please see examples from existing ebuilds. 
https://github.com/gentoo/gentoo/tree/master/acct-group
https://github.com/gentoo/gentoo/tree/master/acct-user

Also check https://wiki.gentoo.org/wiki/Project:Quality_Assurance/UID_GID_Assignment

acct-* ebuilds need to be reviewed in the gentoo-dev mailing list. See example from archive.

Rest looks good.

**Comment by khumarahn** [comment]
I made the changes with acct-user and acct-group. I used UID=170, but I am not sure this is a good number, and how one reserves it.

I also removed write permissions for the i2pd group on config files in `/etc/i2pd`. They are not required by the daemon, and historically they were in place so that a regular user, which is a member of i2pd group, can change the settings. This was non-standard (normally, packages do not do this), and strange: e.g. such a user cannot restart the daemon. Also, not necessary because i2pd can be run as a regular user in the first place.

**Comment by gyakovlev** [comment]
Good that configs no longer writeable by non-root in /etc.
you can check an this http://michael.orlitzky.com/articles/configuration_should_be_owned_and_writable_only_by_root.xhtml for examples why.

As @juippis mentioned , can you please send UID assignment packages for review to gentoo-dev@lists.gentoo.org ?
you can check for recent examples here https://archives.gentoo.org/gentoo-dev/threads/2019-08/
if you don't receive any feedback for 3 of days - it' normal and means it's ok to merge.
note: you need to be subscribed to the list to be able to post.

**Comment by khumarahn** [comment]
Thanks. I sent an email to gentoo-dev.

**Comment by khumarahn** [comment]
My email does not show up on the archives. May it be that I cannot send to gentoo-dev, not being a gentoo developer?

**Comment by khumarahn** [comment]
I corrected a commit message ("UID" -> "GID" for group) and force pushed. There are still no objections to `UID=GID=170`, so this PR is getting ripe.

**Comment by khumarahn** [comment]
Changed UID=GID=170 to 470, because arch linux uses 170 for zabbix.

**Comment by juippis** [comment]
> Changed UID=GID=170 to 470, because arch linux uses 170 for zabbix, and fedora uses 170 for avahi.

Could you update your thread in gentoo-dev with this info? If it doesnt get any objections, I'd say this is ready (will test and merge if no replies in gentoo-dev happen)

**Comment by khumarahn** [comment]
I updated the thread right away.

**Comment by juippis** [comment]
> I updated the thread right away.

Failed to notice that :)

I rebased your commits and updated the latest ebuild a bit, user.eclass does not need to be inherited anymore and 'keepdir' should be handled with acct-user ebuild. I tested these, but if they cause some trouble, let me know. 

Thanks for the contributions!

**Comment by juippis** [comment]
92925dd371befe93729353e3fd45fe442006e64d

**Comment by khumarahn** [comment]
thanks!!

**Comment by khumarahn** [comment]
@juippis do UID/GID need to be added to https://wiki.gentoo.org/wiki/Project:Quality_Assurance/UID_GID_Assignment ?

**Comment by juippis** [comment]
Thanks for the reminded!

**Pull request closed without merging**